cmd/autoprowjobdispatcher/main.go

```diff
@@ -0,0 +1,179 @@
+	fs.StringVar(&o.prometheusPasswordPath, "prometheus-password-path", "", "The path to a file containing the Prometheus password")
+	o.AddFlags(fs)
+	o.AllowAnonymous = true
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	if o.assign == "" {
+		return fmt.Errorf("--assign is mandatory")
+	}
+	if o.prometheusPasswordPath == "" {
+		return fmt.Errorf("--prometheus-password-path cannot be empty string")
+	}
+	return o.GitHubOptions.Validate(false)
+}
+
+func runAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
+
+	logrus.Infof("Running: %s", fullCommand)
+	if err := bumper.Call(stdout, stderr, cmd, args...); err != nil {
+		return false, fmt.Errorf("failed to run %s: %w", fullCommand, err)
+	}
+
+	changed, err := bumper.HasChanges()
+	if err != nil {
+		return false, fmt.Errorf("error occurred when checking changes: %w", err)
+	}
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
+	}
+
+	gitCmd := "git"
+	if err := bumper.Call(stdout, stderr, gitCmd, []string{"add", "."}...); err != nil {
+		return false, fmt.Errorf("failed to 'git add .': %w", err)
+	}
+
+	commitArgs := []string{"commit", "-m", fullCommand, "--author", author}
+	if err := bumper.Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
+		return false, fmt.Errorf("fail to %s %s: %w", gitCmd, strings.Join(commitArgs, " "), err)
+	}
+
+	return true, nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.GitHubOptions.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Failed to start secrets agent")
+	}
+
+	gc, err := o.GitHubOptions.GitHubClient(sa, false)
+	if err != nil {
+		logrus.WithError(err).Fatal("error getting GitHub client")
+	}
+
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	steps := []struct {
+		command   string
+		arguments []string
+	}{
+		{
+			command:   "/usr/bin/prow-job-dispatcher",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs", "--prometheus-password-path", o.prometheusPasswordPath, "--prometheus-username", "ci", "--config-path", "core-services/sanitize-prow-jobs/_config.yaml"},
```

Comment:
The prometheus address is hardcoded into the tool?

Reply:
Make `dispatcher.PrometheusOptions` to share between those 2 tools.